src/core/thread/mle.cpp

```diff
@@ -1362,17 +1362,18 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
 
 #if OPENTHREAD_CONFIG_DUA_ENABLE
     {
-        const Ip6::Address *dua = Get<DuaManager>().GetDomainUnicastAddress();
+        // Cache Domain Unicast Address.
+        address = Get<DuaManager>().GetDomainUnicastAddress();
 
-        if (dua && Get<ThreadNetif>().IsUnicastAddress(*dua))
+        if (Get<ThreadNetif>().IsUnicastAddress(address))
```

Comment:
One concern is that when `DUA` is not added to `Netif`, this address can be any value, i.e. can be equal a non-DUA address. Might incorrectly prioritize a non-DUA address despite a very low chance?

Reply:
For what I can see, there are 3 possible situations for the address
1) valid DUA
2) valid domain prefix, but reserved iid(when generate iid fails)
3) All 0,  unspecified address (when there is no available domain prefix)

For 1), it should be added to Thread interface already;
For 2), first little chance for GenerateIid fails; and even fail, I don't think it will be coincident with other addresses added to the Thread interface - note there should be only one DUA either manually configured or generated according to the spec.
For 3), I don't think there would be such address added to Thread interface.

Any other possible scenarios I missed? @simonlingoogle

Reply:
Submitted a new patch to clear the address when possible so that generally only above 1) and 3) may happen.

Any more concerns? @simonlingoogle

Reply:
My concerns have been resolved with this patch. 

My major concern was actually about 1). Since we are going to have the fixed `DUA` feature, a valid DUA, which is not added to `Netif`, can be set to almost any address. 

We should have no such issue  since now we guarantee that when `DUA` is not in `Netif`, it is always cleared to `::`.